h1st/model/remote_model.py

```diff
@@ -0,0 +1,157 @@
+from typing import Dict, Tuple
+import requests
+from .predictive_model import PredictiveModel
+import pandas as pd
+
+
+class RemoteModel(PredictiveModel):
+    """
+    Class for making inference calls with models served via the Aitomatic Wed API
+
+    Ex: 
+    model = RemoteModel(API_TOKEN, "MyModelName").load_params()
+    predictions = model.predict({'X': MyDataFrame})
+    """
+    PREDICTION_ENDPOINT = ''
+    METADATA_ENDPOINT = ''
+
+    def __init__(self, api_access_token, model_name):
+        """
+        Initialize remote model
+
+        :param api_access_token: Aitomatic API Access Token
+        :param model_name: name of the model being used
+        """
+        self.model_name = model_name
+        self.api_token = api_access_token
+        self.headers = {
+            'API-ACCESS-TOKEN': self.api_access_token,
+            'Content-Type': 'application/json'
+        }
+
+    def predict(self, input_data: Dict) -> Dict:
+        """
+        Logic to generate prediction from data
+
+        :params input_data: input data for prediction, dictionary with data under key 'X'
+        :return: a dictionary with key `predictions` containing the predictions
+        """
+        # Convert data to JSON safe dict
+        json_data, types_dict = convert_data_to_json(input_data)
+
+        # Create web request dicts
+        request_data = {
+            'model_name': self.model_name,
+            'version': self.version,
+            'input_data': json_data
+        }
+
+        # Convert data to json str (NpEncoder allows numpy array conversion)
+        request_data = json.dumps(request_data, cls=NpEncoder)
+
+        # Make web request
+        resp = requests.post(
+            self.PREDICTION_ENDPOINT,
+            headers=self.headers,
+            data=request_data
+        )
+
+        # Handle request errors
+        if resp.status_code != 200:
+            err = f'{resp.status_code}: {resp.content}'
+            raise ConnectionError(err)
+
+        # Convert response back to correct types
+        # predictions format to match input_data['X'] format/types
+        predictions = convert_json_to_data(json.loads(resp.content), types_dict)
```

Comment:
what is the expect format of the api response here? as we understand, the return result will have this format 
```
{
  "result_path": "s3_path_to_store_predict_result",
  "result": "{json_obj_contain_predict_result}"
}
```

Reply:
@phamhoangtuan see the last page of this document: https://docs.google.com/document/d/1AwKNF0h6ZGTfLt9eG9rNlrkFbj0bCbtOZYnwDXuQV1c/edit?usp=sharing

This is just proposed so we can discuss changes, so including "result_path" in the response is fine, and "result" instead of "predictions". I just thought "predictions" to stay consistent with our model outputs. Either way the input json and output json should be convertible to pandas DataFrame or , for predictions, Series if only a single prediction column.

Reply:
this is the sample output from the API
```
{
  "status": "success",
  "result": {
    "predictions": {
      "setosa": {
        "0": 0.632,
        "1": 0.632,
        "2": 0.501,
        "3": 0.501,
        "4": 0.501,
        "5": 0.512,
        "6": 0.504,
        "7": 0.5,
        "8": 0.632,
        "9": 0.519,
        "10": 0.5,
        "11": 0.632,
        "12": 0.632,
        "13": 0.501,
        "14": 0.632,
        "15": 0.512,
        "16": 0.632,
        "17": 0.5,
        "18": 0.507,
        "19": 0.632,
        "20": 0.5,
        "21": 0.632,
        "22": 0.502,
        "23": 0.501,
        "24": 0.632,
        "25": 0.5,
        "26": 0.5,
        "27": 0.5,
        "28": 0.632,
        "29": 0.5
      },
      "veriscolor": {
        "0": 0.632,
        "1": 0.575,
        "2": 0.632,
        "3": 0.632,
        "4": 0.632,
        "5": 0.632,
        "6": 0.632,
        "7": 0.575,
        "8": 0.553,
        "9": 0.632,
        "10": 0.544,
        "11": 0.544,
        "12": 0.611,
        "13": 0.632,
        "14": 0.599,
        "15": 0.632,
        "16": 0.564,
        "17": 0.632,
        "18": 0.632,
        "19": 0.622,
        "20": 0.536,
        "21": 0.622,
        "22": 0.632,
        "23": 0.622,
        "24": 0.575,
        "25": 0.587,
        "26": 0.622,
        "27": 0.632,
        "28": 0.632,
        "29": 0.599
      }
    }
  },
  "result_file_path": "s3://saas-aito-dev-projects/48379e64-041d-4789-8932-5043f4f623bc-Iris/inference/a0a8d0e5-1f4b-46bc-a00e-d0e66a924e7f"
}
```
we can get `prediction` from `result` obj